Assets bulk edit (#901)
akeneo/pim-helpcenter: content/md/products-super-power/work-on-your-assets.md

related: create-and-display-assets
---

# Enrich your assets

> ## Enrich an asset

Reviewer: ```suggestion
## Enrich a single asset
```

> Once you have created your asset, you can enrich it in order to add some information.

Reviewer: ```suggestion
Once you have created your asset, you can enrich it.
```

> Nothing simpler: just go on an asset and add/change the values you want.  

Reviewer: ```suggestion
It's pretty straightforward, you just have to go to an asset and add/change the values you want.  
```

![Assets Locale Switcher](../img/Assets_LocaleSwitcher.png)

## Bulk edit a selection of asset attributes
It is possible to bulk edit your own selection of asset attributes within an asset family.
Let's imagine that you need to add the same tag to several assets, for instance, their main color attribute. Instead of manually adding the tag for each one of these assets, you can easily do it once for your selection of assets, thanks to a bulk action. Here's how to do do:

1. Check the box under each asset you would like to bulk edit
1. In the bottom bar, click on `edit`
1. At the "edit" step, click on `add attributes` in the top right menu
![Assets attributes selection](../img/Assets_bulk-edit-add-attributes.png)
1. Select the attributes you would like to bulk edit
![Assets attributes selection](../img/Assets_bulk-edit-attributes-selection.png)
1. Fill in the values you want for each selected attribute
1. Once you are done, click on `next` in the top right corner
1. At the "confirm" step, you can visualise the changes that will be made. If you want to continue, click on `Confirm`

::: warning

> Please note that it is possible to add the same attribute multiple times. For instance, if you would like to edit an attribute that is scopable and/or localisable, you will have the possibility to do so by adding the attribute as many times as you want, but you will have to carefully select a different locale and/or scope for each of them so that you will avoid to setting different values for the same attribute with the same locale and/or same scope.

Reviewer 2: ```suggestion
Please note that it is possible to add the same attribute multiple times. For instance, if you would like to edit an attribute that is scopable and/or localisable, you will have the possibility to do so by adding the attribute as many times as you want, but you will have to carefully select a different locale and/or scope for each of them, if it's not the case, you are going to have an error when clicking on `next`.
```

:::

> 

Reviewer 2: ```suggestion
::: warning
Attribute read only in the UI and target of transformation cannot be selected in the bulk edit
```

Author: Good point, thanks!

> 

Reviewer 2: Maybe add permission requirement for mass edit asset

Author: Thanks for the clarification!

![Assets Locale Switcher](../img/Assets_LocaleSwitcher.png)

> ## Bulk edit a selection of asset attributes

Reviewer: ```suggestion
## Bulk edit multiple assets
```

> If you have the edit permission on an asset family, it is possible to bulk edit your own selection of asset attributes within this asset family.

Reviewer: ```suggestion
If you have the `edit` permission on an asset family, it is possible to bulk edit your own selection of assets within this asset family.
```

> Let's imagine that you need to add the same tag to several assets, for instance their main color attribute. Instead of manually adding the tag for each one of these assets, you can easily do it in one action for your selection of assets, thanks to a bulk edition. Here's how to do so:

Reviewer: ```suggestion
Let's imagine that you need to add the same tag to several assets, for instance, their main color attribute. Instead of manually adding the tag for each of these assets, you can easily do it in one action for your selection of assets, thanks to a bulk edition. Here's how to do so:
```

> 1. Check the box under each asset you would like to bulk edit

Reviewer: ```suggestion
1. Check the box under each asset you want to bulk edit
```

1. In the bottom bar, click on `edit`

> 1. At the "edit" step, click on `add attributes` in the top right menu

Reviewer: ```suggestion
1. In the bulk edit pop-in, click on `add attributes` in the top-right menu
```

> ![Assets attributes selection](../img/Assets_bulk-edit-add-attributes.png)

Reviewer: ```suggestion
![Asset attributes selection](../img/Assets_bulk-edit-add-attributes.png)
```

> 1. Select the attributes you would like to bulk edit

Reviewer: ```suggestion
1. Select the attributes you want to bulk edit
```

> ![Assets attributes selection](../img/Assets_bulk-edit-attributes-selection.png)

Reviewer: ```suggestion
![Asset attributes selection](../img/Assets_bulk-edit-attributes-selection.png)
```

1. Fill in the values you want for each selected attribute

> 1. Once you are done, click on `next` in the top right corner

Reviewer: ```suggestion
1. Once you are done, click on `next` in the top-right corner
```

> 1. At the "confirm" step, you can visualise the changes that will be made. If you want to continue, click on `Confirm`

Reviewer: ```suggestion
1. You can then review the changes that will be made. If you want to continue, click on `Confirm`
```

::: warning

> Please note that it is possible to add the same attribute multiple times. For instance, if you would like to edit an attribute that is scopable and/or localisable, you will have the possibility to do so by adding the attribute as many times as you want. However you will have to carefully select a different locale and/or scope for each of them, in order to avoid displaying an error when clicking on `Next`.

Reviewer: ```suggestion
Please note that it is possible to add the same attribute multiple times. For instance, if you want to edit an attribute that has a different value per channel and/or per locale, you have the possibility to do so by adding the attribute as many times as you want. However, you have to carefully select a different locale and/or channel for each of them, so that you will avoid setting different values for the same attribute with the same locale and/or channel.

```

:::

::: info

> The attributes that are configured as "read only in the UI" cannot be selected in the bulk edit.

Reviewer: ```suggestion
The attributes that are configured as "read-only in the UI",  and those that are used as target attributes for an asset transformation cannot be edited in bulk. Check out our article to find out more [about transformations and target attributes](https://help.akeneo.com/pim/serenity/articles/assets-transformation.html).
```

![Assets Locale Switcher](../img/Assets_LocaleSwitcher.png)

## Bulk edit multiple assets

> If you have the `edit` permission on an asset family, it is possible to bulk edit your own selection of assets within this asset family.

Reviewer 3: ```suggestion
If you have the `edit` permission on an asset family, you can bulk edit attributes from one or more assets from the same asset family.
```

> Let's imagine that you need to add the same tag to several assets, for instance, their main color attribute. Instead of manually adding the tag for each of these assets, you can easily do it in one action for your selection of assets, thanks to a bulk edition. Here's how to do so:

Reviewer 3: Hi @felix-couprie your example here is to add a tag that would contain the main color of the image right? Otherwise, can we be more explicit with the use case? "main color attribute" could be misleading.
Example given below

Reviewer 3: ```suggestion
Let's imagine you need to add a tag to several assets from the same family, this tag will give information on the main color of the image, and this color is the same for all your selected assets. Instead of manually adding the tag for each of these assets, you can easily do it in one action for your selection of assets, thanks to a bulk edition. Here's how to do so:
```

Author: Thanks Laure, it sounds clearer!

1. Check the box under each asset you want to bulk edit
1. In the bottom bar, click on `edit`
1. In the bulk edit pop-in, click on `add attributes` in the top-right menu
![Asset attributes selection](../img/Assets_bulk-edit-add-attributes.png)
1. Select the attributes you want to bulk edit
![Asset attributes selection](../img/Assets_bulk-edit-attributes-selection.png)
1. Fill in the values you want for each selected attribute
1. Once you are done, click on `next` in the top-right corner
1. You can then review the changes that will be made. If you want to continue, click on `Confirm`

::: warning
Please note that it is possible to add the same attribute multiple times. For instance, if you want to edit an attribute that has a different value per channel and/or per locale, you have the possibility to do so by adding the attribute as many times as you want. However, you have to carefully select a different locale and/or channel for each of them, so that you will avoid setting different values for the same attribute with the same locale and/or channel.

:::

::: info

> The attributes that are configured as "read-only in the UI",  and those that are used as target attributes for an asset transformation cannot be edited in bulk. Check out our article to find out more [about transformations and target attributes](https://help.akeneo.com/pim/serenity/articles/assets-transformation.html).

Reviewer 3: ```suggestion
The attributes configured as "read-only in the UI" and those used as target attributes for an asset transformation cannot be edited in bulk. Read our article to learn more about [transformations and target attributes](https://help.akeneo.com/pim/serenity/articles/assets-transformation.html).
```

The completeness is **automatically calculated** as soon as we enrich a required attribute. It means that the bullet point will disappear as soon as the required attribute is filled, and same for the completeness percentage evolution.

# Edit assets
## Bulk edit a selection of asset attributes

> It is possible to bulk edit your own selection of asset attributes in an asset family.

Reviewer 4: Je pense qu'on pourrait rajouter une phrase d'intro en donnant un exemple de use-case. Du genre : 
Let's imagine that you need to add the same tag to several assets, for instance, their main color. Instead of manually adding the tag for each of these assets, you can easily do it once for your selection of assets, via a bulk action. Let's see how to:

1. Check the box under each asset you would like to bulk edit
1. In the bottom bar, click on `edit`
1. At the "edit" step, click on `add attributes` in the top right menu

> ![Assets attributes selection](../img/Assets_bulk-edit-add-attributes.png)

Reviewer 4: Attention, sur ce screenshot, les textes des liens vers le helcenter sont inversés, faudrait que les gars redéploient la version corrigée

1. Select the attributes you would like to bulk edit

> ![Assets attributes selection](../img/Assets_bulk-edit-attributes-selection.png)

Reviewer 4: l'image ne s'affiche pas

1. Fill in the values you want for each selected attribute
1. Once you are done, click on `next` in the top right corner
1. At the "confirm" step, you can visualise the changes that will be made. If you want to continue, click on `Confirm`

::: warning

> Please note that it is possible to add the same attribute multiple times. For instance, if you would like to edit an attribute that is scopable and/or localisable, you will have the possibility to do so by adding the attribute as many times as you want, but you will have to carefully select a different locale and/or scope for each of them so that you will avoid to setting different values for the same attribute with the same locale and/or same scope.

Reviewer 4: ```suggestion
Please note that it is possible to add the same attribute multiple times. For instance, if you would like to edit an attribute that has a different value per channel and/or per locale, you will have the possibility to do so by adding the attribute as many times as you want, but you will have to carefully select a different locale and/or channel for each of them so that you will avoid setting different values for the same attribute with the same locale and/or same channel.
```

Reviewer 4: Je pense qu'il faudrait simplifier la phrase mais ça sort de mes compétences :D (à voir avec Camille)
